quinn-proto/src/connection/mod.rs

```diff
@@ -1324,8 +1324,13 @@ where
         }
 
         // Update state for confirmed delivery of frames
-        for (id, _) in info.retransmits.reset_stream {
-            self.streams.reset_acked(id);
+        match &info.retransmits {
```

Comment:
Can we do something like `with_retransmits()` that covers this case and the one in `ensure_probe_queued()`?

Reply:
I agree, it's probably cleaner. But then we are at the `Retransmits` and `InnerRetransmits` type, or maybe `Retransmits` and `OptionalRetransmits`, and I thought @Ralith wasn't too much of a fan.

I guess I can live with `struct OptionalRetransmits(Option<Box<Retransmits>>)`

Reply:
I appreciate being able to express that something *isn't* optional. I don't have any objection to having both. `ThinRetransmits` could also be a good name.